api/base_controller.go

```diff
@@ -149,6 +156,22 @@ func (c *BaseController) DeleteObjects(r *web.Request) (*web.Response, error) {
 	log.C(ctx).Debugf("Deleting %ss...", c.objectType)
 
 	criteria := query.CriteriaForContext(ctx)
+	idCriterion := getIDCriterion(criteria)
+
+	objectList, err := c.repository.List(ctx, c.objectType, idCriterion...)
+	if err != nil {
+		return nil, err
+	}
+
+	log.C(ctx).Debugf("Attempting to validate deletion of %s object(s)", c.objectType)
+	for i := 0; i < objectList.Len(); i++ {
+		log.C(ctx).Debugf("Attempting to validate deletion of %s object with ID (%s)", c.objectType, objectList.ItemAt(i).GetID())
+		err = c.ResourceValidator.ValidateDelete(ctx, c.repository, objectList.ItemAt(i))
```

Comment:
Should we first validate all objects and then exit with all errors or like it is know - fail fast? This is just an idea

Reply:
It's surely an idea. It will however increase the amount of DB calls and because of that also increase the chance of race conditioning. Let's discuss it with the team